lib/services/http-service.js

```diff
@@ -62,94 +62,82 @@ di.annotate(httpServiceFactory,
-            req._startAt = process.hrtime();
-            res.locals.ipAddress = remoteAddress(req);
-            res.locals.scope = ['global'];
+    HttpService.prototype.start = function() {
+        var self = this;
+        // Create Http Proxy
+        if (this.endpoint.proxiesEnabled) {
+            var httpProxies = configuration.get('httpProxies');
+            if (httpProxies && _.isArray(httpProxies)) {
+                _.forEach(httpProxies, createHttpProxy.bind(this, self.app));
+            }
+        }
 
-            onFinished(res, function () {
-                if (!req._startAt) {
-                    return '';
-                }
+        if (this.endpoint.httpsEnabled) {
+            this.server = https.createServer(this.httpsOptions, this.app);
 
-                var diff = process.hrtime(req._startAt),
-                    ms = diff[0] * 1e3 + diff[1] * 1e-6;
+            wss.start(this.server, {
+                secureOptions: this.httpsOptions
+            });
+        } else {
+            this.server = http.createServer(this.app);
 
-                var data = {
-                    ipAddress: res.locals.ipAddress
-                };
+            wss.start(this.server);
+        }
 
-                if (res.locals.identifier) {
-                    data.id = res.locals.identifier;
-                }
+        this.server.on('close', function() {
+            logger.info('Server Closing.');
+        });
 
-                logger.debug(
-                    'http: ' + req.method +
-                    ' ' + res.statusCode +
-                    ' ' + ms.toFixed(3) +
-                    ' - ' + req.originalUrl,
-                    data
-                );
-
-                eventsProtocol.publishHttpResponse(
-                    res.locals.identifier || 'external',
-                    {
-                        address: res.locals.ipAddress,
-                        method: req.method,
-                        url: req.originalUrl,
-                        statusCode: res.statusCode,
-                        time: ms.toFixed(3)
-                    }
-                );
-            });
+        return this.server.listen(this.endpoint.port, this.endpoint.address);
+    };
 
-            lookupService.ipAddressToNodeId(res.locals.ipAddress).then(function (nodeId) {
-                res.locals.identifier = nodeId;
-            }).catch(Errors.NotFoundError, function () {
-                // No longer log NotFoundErrors
-            }).catch(function (error) {
-                events.ignoreError(error);
-            }).finally(function () {
-                next();
-            });
-        }
+    HttpService.prototype.stop = function() {
+        wss.stop();
+        return this.server.close();
+    };
 
+    HttpService.prototype._setup =  function() {
+        var app = this.app;
+        var endpoint = this.endpoint;
         //Swagger support
         var swaggerConfig = {
```

Comment:
@WangWinson Could you update the diff with this change?  I think we should use the same approach as with V1: add a `swagger` property to each endpoint to configure which swagger file to use, but for now configure only a single endpoint with a single swagger file.